tasty-query/shared/src/main/scala/tastyquery/Types.scala

```diff
@@ -484,20 +495,27 @@ object Types {
 
     /** The class symbol of this type, None if reduction is not possible */
     @tailrec
-    private[tastyquery] final def classSymbol(using Context): Option[ClassSymbol] = this.widen match
+    private[tastyquery] final def classSymbol(using Context): Option[ClassSymbol] = this match
       case TypeRef.OfClass(cls) =>
         Some(cls)
       case tpe: TypeRef =>
         tpe.optAliasedType match
           case Some(alias) => alias.classSymbol
           case None        => None
+      case tpe: TermRef =>
+        tpe.underlyingOrMethodic match
+          case underlying: Type => underlying.classSymbol
+          case _: MethodicType  => None
       case tpe: TypeProxy =>
         tpe.superType.classSymbol
       case tpe: TypeLambda =>
         // apparently we need this :(
         tpe.resultType.classSymbol
-      case _ =>
+      case _: NothingType | _: AnyKindType | _: OrType | _: AndType =>
```

Comment:
I would like to see that `(Array[String] with Array[String]) with Array[String]` in a scala 2 source still erases correctly

Reply:
it looks like these parents will be flattened though so it should be fine

Reply:
We don't compile any test source using Scala 2, so it's a bit difficult to this from source. I added a direct unit test of erasure instead.